subsys/llext/llext_link.c

```diff
@@ -116,16 +116,27 @@ static void llext_link_plt(struct llext_loader *ldr, struct llext *ext,
 		const char *name = llext_string(ldr, ext, LLEXT_MEM_STRTAB, sym_tbl.st_name);
 
 		/*
-		 * Both r_offset and sh_addr are addresses for which the extension
-		 * has been built.
+		 * FIXME: The calculations below assumes offsets from the
+		 * beginning of the .text section in the ELF file can be
+		 * applied to the memory location of mem[LLEXT_MEM_TEXT].
+		 * This is possible only when peek() is used, and
+		 * does also require careful linking.
 		 */
-		size_t got_offset;
-
-		if (tgt) {
-			got_offset = rela.r_offset + tgt->sh_offset -
+		size_t offset;
+
+		if (ldr->hdr.e_type == ET_REL) {
```

Comment:
isn't it this exact location that we discussed before and that cannot work for SOF because there we perform symbol linking on an image in a temporary storage, i.e. addresses where it's stored aren't the same as where it will be executed?

Reply:
Yes, maybe you are right, I totally forgot (... as there is no comment and no test whatsoever for that scenario). I just took the opportunity to do away with the "looking up sections by name" stuff, but I will leave the Xtensa paths exactly as before.